### PR TITLE
ODC-7316: Add discoverability of the IDE extensions from Create Serverless form

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Story**: https://issues.redhat.com/browse/ODC-7316

**Description**:

Added below IDE extensions in create serverless form,

VSCode Knative IDE extension: https://marketplace.visualstudio.com/items?itemName=redhat.vscode-knative
IntelliJ Knative Plugin: https://plugins.jetbrains.com/plugin/16476-knative--serverless-functions-by-red-hat

**Screenshots**:
Screenshot 2023-05-24 at 2 37 05 PM

https://github.com/openshift/console/assets/102503482/a843a1d3-b7c2-4a5a-b789-710b0155ecc3

### CODE REVIEW

**Comment by Lucifergene** [comment]
@lokanandaprabhu Can you please share a video demonstrating the effect of the cards on gently reducing the screen's width? 
For this, you can adjust the width of the Chrome Dev Tools Side Panel.

I think this will help to understand the alignments and positions of the cards on different screens.

**Comment by lokanandaprabhu** [comment]
Hi @Lucifergene ,

Added the video. The cards can be visible only for 'lg' and 'xl'.

**Comment by Lucifergene** [comment]
Imo, this looked good as the position wouldn't change much on resizing the screen:

image

But, if the current design is finalized, it's still okay :+1:

**Comment by vikram-raj** [comment]
font-weight of `provided by` text is more than the title. Can you please change it to a normal font-weight?
image

**Comment by vikram-raj** [comment]
@lokanandaprabhu `Create` and `Cancel` buttons are hanging in between the screen which looks odd. Can we move it to the bottom or remove the extra spaces between the input field and buttons? 

And the `Provided by` text still looks similar to the title, need to remove the title font family. See the recording. 

Check Operator hub cards for reference.

https://github.com/openshift/console/assets/2561818/bf7ebf92-c2b8-4f87-a466-55c889a70d73

**Comment by vikram-raj** [comment]
@lokanandaprabhu 
- Font style now looks good
- I am not sure if we can go with the non-sticky footer. It will become inconsistent with other forms we have in the console.  wdyt @jerolimov ?

File: frontend/public/imgs/intellij.png
Comment (jerolimov) [inline frontend/public/imgs/intellij.png:1]: File should be at least part of the imgs/logos folder, or in the same folder as the card component.

File: frontend/public/components/catalog/catalog-item-icon.tsx [hunk 1]
```diff
@@ -100,6 +100,8 @@ import * as wordpressImg from '../../imgs/logos/wordpress.svg';
 import * as xamarinImg from '../../imgs/logos/xamarin.svg';
 import * as zendImg from '../../imgs/logos/zend.svg';
 import * as operatorImg from '../../imgs/logos/operator.svg';
+import * as vscodeImg from '../../imgs/vscode.png';
+import * as intellijImg from '../../imgs/intellij.png';
```
Comment (jerolimov) [inline frontend/public/components/catalog/catalog-item-icon.tsx:104]: The other logos was part of the logos folder. I'm also unsure if we want add these both to the catalog list.. Its not wrong, but also not necessary.

File: frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx [hunk 2]
```diff
@@ -61,6 +72,8 @@ const AddServerlessFunctionForm: React.FC<
     image,
   } = values;
   const isPipelineEnabled = useFlag(FLAG_OPENSHIFT_PIPELINE);
+  const imgSrcVSCode = getImageForIconClass('icon-vscode');
+  const imgScrIntelliJ = getImageForIconClass('icon-intellij');
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx:76]: You can import logos also like this:

```ts
import * as vscodeIcon from './icon-vscode.png';
```

and use it like this:

```ts
<img src={vscodeIcon} 
```

The `getImageForIconClass` is to get dynamic images.

**Review by jerolimov — CHANGES_REQUESTED** [review]

**Comment by jerolimov** [comment]
> @lokanandaprabhu
> 
>     * Font style now looks good
> 
>     * I am not sure if we can go with the non-sticky footer. It will become inconsistent with other forms we have in the console.  wdyt @jerolimov ?

I agree with Vikram, we should accept that its non sticky just because we add two icons on the right hand side.

File: frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx [hunk 3]
```diff
@@ -120,44 +139,117 @@ const AddServerlessFunctionForm: React.FC<
     }
   }, [image, isPipelineEnabled]);
 
+  const ExtensionCard: React.FC<ExtensionCardProps> = ({ icon, link, title, description }) => {
+    return (
+      <Card>
+        <CardTitle>
+          <div className="odc-serverless-extensions-card-logo">
+            <div className="odc-serverless-extensions-card-logo__icon">
+              <span className="co-catalog-item-icon__bg">
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx:148]: Please don't reuse classes from other plugins.

File: frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx [hunk 4]
```diff
@@ -120,44 +139,117 @@ const AddServerlessFunctionForm: React.FC<
...
-          </>
-        )}
-        {validated !== ValidatedOptions.default &&
-          strategy !== ServerlessBuildStrategyType.ServerlessFunction && (
-            <Alert
-              variant="warning"
-              isInline
-              title={t('devconsole~func.yaml is not present or builder strategy is not s2i')}
+      <Flex direction={{ default: 'column', sm: 'row' }}>
+        <FlexItem flex={{ default: 'flex_1' }} alignSelf={{ default: 'alignSelfFlexStart' }}>
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/serverless-function/AddServerlessFunctionForm.tsx:177]: Screenshot from 2023-06-15 09-10-16

Esp. with the light theme I notice these different gaps between the boxes.

You can remove the default spacing from the Flex components with the `spaceItems` props:

```suggestion
      <Flex direction={{ default: 'column', sm: 'row' }} spaceItems={{ default: 'spaceItemsNone' }}>
        <FlexItem flex={{ default: 'flex_1' }} alignSelf={{ default: 'alignSelfFlexStart' }}>
```

Wdyt?

**Review by jerolimov — CHANGES_REQUESTED** [review]
Hi, this looks overall good to me. But I have some small issues I hope you can address:

1. Avoid `!important` and use PF variables instead of hard coded px sizes.
2. The spacing around the cards looks could be improved (see comments below)
3. There are some small optional improvements in the scss file.

**Comment by lokanandaprabhu** [comment]
Propagate labels from the epic:

/label docs-approved
/label px-approved

**Review by jerolimov — APPROVED** [review]
Propage labels from the epic:

/label docs-approved
/label px-approved

Thanks for adding one additional e2e test.

/label qe-approved

**Review by jerolimov — APPROVED** [review]
/lgtm
/approve

**Pull request merged**